test/functional/cursor_tests.js

```diff
@@ -4382,4 +4382,29 @@ describe('Cursor', function() {
       }
     }
   );
+
+  it('should return false when exhausted and hasNext called more than once', function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient({ w: 1 }, { poolSize: 1, auto_reconnect: false });
+
+    client.connect(function(err, client) {
```

Comment:
if you wrote this all as promises you could avoid the need to use `done` completely, by returning the promise chain. Mocha is smart enough to handle this.

Reply:
You'd still need a way to properly teardown the cursor and client.

Reply:
sure, but that doesn't preclude the fact that you can return promises?